### PR TITLE
added delete, head and patch methods

### PR DESCRIPTION
**Pull request by stearm**
works on #9

### CODE REVIEW

**Comment by mcollina** [comment]
@delvedor what do you think? Where you working on something similar?

**Comment by delvedor** [comment]
Yes, I've finished to implement  `'DELETE'`, `'HEAD'`, `'PATCH'` and `'OPTIONS'` methods, but in a different way.
I'm using an if/else statement for performances, and I'm not checking for the body in the delete request, since is not needed.

File: fastify.js [hunk 1]
```diff
@@ -96,14 +111,32 @@ function build () {
...
+          case 'DELETE':
+            if (req.headers['content-type'] === 'application/json') {
+              jsonParser(req, bodyParsed(handle, params, req, res))
+            } else {
+              res.statusCode = 415
+              res.end()
+            }
+            break
+          default:
+            throw Error(`Invalid method: ${req.method}`)
```
Comment (delvedor) [inline fastify.js]: This will throw at runtime!!

Comment (delvedor) [inline fastify.js]: The default should be a 404 or 400 :)

Comment (stearm) [inline fastify.js]: 👍

File: examples/example.js [hunk 2]
```diff
@@ -22,6 +22,15 @@ fastify
   .post('/', schema, function (req, reply) {
     reply(null, { hello: 'world' })
   })
+  .head('/', schema, function (req, reply) {
+    reply(null, { hello: 'world' })
```
Comment (delvedor) [inline examples/example.js]: This example is not completely correct, head should not return a body, is just to get the headers without download the body too :)

Comment (stearm) [inline examples/example.js]: Oh yes, sorry copy&past typo. I can fix these things if you want, or only help you on others things!

File: test/head.test.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+'use strict'
+
+const t = require('tap')
+require('./helper').payloadMethod('patch', t)
```
Comment (delvedor) [inline test/head.test.js]: I guess this should be `'head'`

**Comment by stearm** [comment]
> I'm not checking for the body in the delete request, since is not needed.
Why? Delete method accepts a body.

**Comment by delvedor** [comment]
> A payload within a DELETE request message has no defined semantics;
   sending a payload body on a DELETE request might cause some existing
   implementations to reject the request.

[RFC](https://tools.ietf.org/html/rfc7231#section-4.3)
[Wikipedia](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Summary_table)

@mcollina your opinion about this?

**Comment by stearm** [comment]
I think it should only be ignored, but it is not forbidden to have a body inside a delete request :)

**Comment by delvedor** [comment]
Indeed, is what I'm saying :)
Since the body is not needed, we not need [this](https://github.com/stearm/fastify/blob/6191d5362f75a38e759a81604b2abfe9d161d061/fastify.js#L130-L137).

**Comment by mcollina** [comment]
IMHO we should not parse the body.

**Comment by stearm** [comment]
Ok no problem :)

**Pull request merged**